core/arch/arm/kernel/trace_ext.c

```diff
@@ -28,20 +28,39 @@
 #include <trace.h>
 #include <console.h>
 #include <kernel/thread.h>
+#include <kernel/spinlock.h>
+#include <mm/core_mmu.h>
 
 const char trace_ext_prefix[] = "TEE-CORE";
 int trace_level = TRACE_LEVEL;
+static unsigned int puts_lock = SPINLOCK_UNLOCK;
 
 void trace_ext_puts(const char *str)
 {
+	uint32_t itr_status = thread_mask_exceptions(THREAD_EXCP_FOREIGN_INTR);
```

Comment:
@MrVan, could you push back a fix with `THREAD_EXCP_ALL` instead of `THREAD_EXCP_FOREIGN_INTR` ?
Wtih this change: `Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

I think it could be nice to have this in the next OP-TEE 2.4.0. Do you think you can update this PR in the short term ?